proto/cosmos/ics23/v1/proofs.proto

```diff
@@ -163,6 +163,10 @@ message ProofSpec {
   int32 max_depth = 3;
   // min_depth (if > 0) is the minimum number of InnerOps allowed (mainly for fixed-depth tries)
   int32 min_depth = 4;
+  // prehash_compare_key is a flag that indicates whether to use the
+  // prehash_key specified by LeafOp to compare lexical ordering of keys for
+  // non-existence proofs.
+  bool prehash_compare_key = 5;
```

Comment:
I think a clearer name might be helpful here similar to our review of #88.

So effectively the database is providing an interface to the application: `store.Set(key, value)`

and underneath the hood it is merklizing this pair by storing the hash of the application-provided key as the key in the merkle tree.

Thus, we liked the nomenclature: `appKey` and `treeKey`.

The `appKey` is the key known to the application, the `treeKey` is the key stored in the tree.

In SMT, the treeKey is the hash of the appKey, while in iAVL they are the same.

So I think `prehash_app_key` would be clearer here as a name. And we can use the docs here to explain there may be a difference between appKey and treeKey for some  trees

Reply:
This is a bit of a bikeshed, but if there's going to be a distinction between `app_key` and `tree_key`, wouldn't it be even clearer to call the field `compare_tree_key`?  Then `compare_tree_key = false` means it compares the app key, and `compare_tree_key = true` means it compares the tree key.

Reply:
Ok now that I understand more of this field.

Perhaps we can call it: `prehash_key_before_comparison`

The way I read the current field name at the moment, I expect it to be a HashOp type not a boolean.

cc: @colin-axner